Python/flowgraph.c

```
@@ -21,6 +21,15 @@
         return ERROR;       \
     }
 
+#define RETURN_IF_FOLD_FAIL(X) \
```

This is only used once so I don't think we should inline it.

---

It is used once here, but I suspect it will be useful when other foldings are migrated as well (unop, binop for example).

---

I don't think this is a good pattern to have. It is too implicit in how it swallows exceptions. Even more so when it's a macro that has no knowledge of the context it's used in.

---

Also, `BINARY_SUBSCR` will be a `BINARY_OP` with oparg `NB_SUBSCR` soon so they will be handled together.

---

This piece of logic will be used in every folding case. What do you suggest to do with it?

---

Deal with that problem when we have it.

---

Swallowing exceptions is a bad thing, but we should do it during BINOP optimization. I guess there will not be any other place where we must swallow exceptions, so I agree with Irit's opinion on this.

---

Not also that `RETURN_IF_ERROR` does *not* swallow exceptions. It is asking for trouble to put a `RETURN_IF_*` next to it which does.

---

Okay, lets deal with it when time comes